blockchain/version.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2013-2016 The btcsuite developers
+// Copyright (c) 2015-2016 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package blockchain
+
+func (b *BlockChain) CalcNextStakeVersion() (uint32, error) {
+	b.chainLock.RLock()
+	defer b.chainLock.RUnlock()
+
+	return b.calcNextStakeVersion(b.bestNode)
+}
+
+func (b *BlockChain) calcNextStakeVersion(node *blockNode) (uint32, error) {
+
+	if node == nil {
+		return 0, ruleError(ErrVersionVoting, "invalid node")
```

Comment:
Should be an `AssertError` because this should never happen.  It is an internal failure in the code as opposed to a consensus failure which is what `ruleError` describes.